internal/output/output_result.go

```diff
@@ -0,0 +1,538 @@
+				count.FixableCount.UnFixed += 1
+			} else {
+				count.FixableCount.Fixed += 1
+			}
+		}
+	}
+}
+
+func getVulnList(vulnMap map[string]OutputVulnResult) []OutputVulnResult {
+	vulnList := make([]OutputVulnResult, 0, len(vulnMap))
+	for _, vuln := range vulnMap {
+		vulnList = append(vulnList, vuln)
+	}
+
+	// Sort projectResults to ensure consistent output
+	slices.SortFunc(vulnList, func(a, b OutputVulnResult) int {
+		return cmp.Compare(a.ID, b.ID)
+	})
+
+	return vulnList
+}
+
+// getNextFixVersion returns the lowest fixed version for a given package and
+// its current installed version, considering the affected ranges. If no fix is
+// available, it returns "No fix available".
+func getNextFixVersion(allAffected []models.Affected, installedVersion string, installedPackage string, ecosystem models.Ecosystem) string {
+	ecosystemPrefix := models.Ecosystem(strings.Split(string(ecosystem), ":")[0])
+	vp, err := semantic.Parse(installedVersion, ecosystemPrefix)
+	if err != nil {
+		return VersionUnsupported
+	}
+
+	minFixVersion := UnfixedDescription
+	for _, affected := range allAffected {
+		if affected.Package.Name != installedPackage || affected.Package.Ecosystem != ecosystem {
+			continue
+		}
+		for _, affectedRange := range affected.Ranges {
+			for _, affectedEvent := range affectedRange.Events {
+				// Skip if it's not a fix version event or the installed version is greater than the fix version.
+				if affectedEvent.Fixed == "" || vp.CompareStr(affectedEvent.Fixed) > 0 {
+					continue
+				}
+
+				// Find the minium fix version
+				if minFixVersion == UnfixedDescription || semantic.MustParse(affectedEvent.Fixed, ecosystemPrefix).CompareStr(minFixVersion) < 0 {
+					minFixVersion = affectedEvent.Fixed
+				}
+			}
+		}
+	}
+
+	return minFixVersion
+}
+
+// getPackageFixedVersion determines the highest version that resolves the most known vulnerabilities for a package.
+func getPackageFixedVersion(ecosystem string, allVulns []OutputVulnResult) string {
+	ecosystemPrefix := models.Ecosystem(strings.Split(ecosystem, ":")[0])
+	maxFixVersion := ""
+	var vp semantic.Version
+	for _, vuln := range allVulns {
+		if vuln.FixedVersion == VersionUnsupported {
+			return UnfixedDescription
+		}
+
+		// Skip vulnerabilities without a fixed version.
+		if vuln.FixedVersion == UnfixedDescription {
+			continue
+		}
+
+		if maxFixVersion == "" {
+			maxFixVersion = vuln.FixedVersion
+			vp = semantic.MustParse(maxFixVersion, ecosystemPrefix)
+
+			continue
+		}
+
+		// Update if the current vulnerability's fixed version is higher
+		if vp.CompareStr(vuln.FixedVersion) < 0 {
+			maxFixVersion = vuln.FixedVersion
+			vp = semantic.MustParse(maxFixVersion, ecosystemPrefix)
+		}
+	}
+
+	// Default to UnfixedDescription if no fix version is found.
+	if maxFixVersion == "" {
+		maxFixVersion = UnfixedDescription
+	}
+
+	return maxFixVersion
+}
+
+func updateVulnCount(original OutputVulnCount, newAdded OutputVulnCount) OutputVulnCount {
```

Comment:
A cleaner way to do this add is to make these member methods on each of the types, and just name it Add(). 

It looks like we are also just assigning the output back onto `original`, so we can even take it by pointer and update in place, though I can see that being less flexible compared to returning and assigning manually.